packages/grid/x-data-grid/src/tests/rows.DataGrid.test.tsx

```diff
@@ -244,46 +247,116 @@ describe('<DataGrid /> - Rows', () => {
-      expect(screen.queryAllByRole('menu')).to.have.length(1);
+      expect(screen.queryAllByRole('menu')).to.have.length(2 + 1);
 
-      fireEvent.click(screen.getAllByRole('button', { name: 'more' })[1]);
+      fireEvent.click(screen.getAllByRole('menuitem', { name: 'more' })[1]);
       clock.runToLast();
-      expect(screen.queryAllByRole('menu')).to.have.length(1);
+      expect(screen.queryAllByRole('menu')).to.have.length(2 + 1);
     });
 
     it('should allow to move focus to another cell with the arrow keys', () => {
       render(
-        <div style={{ width: 300, height: 300 }}>
-          <DataGrid
-            rows={[{ id: 1, name: 'John' }]}
-            columns={[
-              { field: 'name' },
-              {
-                field: 'actions',
-                type: 'actions',
-                getActions: () => [<GridActionsCellItem icon={<span />} label="print" />],
-              },
-            ]}
-          />
-        </div>,
+        <TestCase getActions={() => [<GridActionsCellItem icon={<span />} label="print" />]} />,
       );
       const firstCell = getCell(0, 0);
-      fireEvent.mouseUp(firstCell);
-      fireEvent.click(firstCell);
-
+      firstCell.focus();
       expect(getActiveCell()).to.equal('0-0');
+
       fireEvent.keyDown(firstCell, { key: 'ArrowRight' });
-      expect(getActiveCell()).to.equal('0-1');
+      const printButton = screen.queryByRole('menuitem', { name: 'print' });
+      expect(printButton).toHaveFocus();
 
-      const button = screen.queryByRole('button', { name: 'print' });
-      button.focus();
-      fireEvent.keyDown(button, { key: 'ArrowLeft' });
+      fireEvent.keyDown(printButton, { key: 'ArrowLeft' });
+      expect(getActiveCell()).to.equal('0-0');
+    });
+
+    it('should focus the first item when opening the menu', () => {
+      render(
+        <TestCase
+          getActions={() => [
+            <GridActionsCellItem icon={<span />} label="print" showInMenu />,
+            <GridActionsCellItem icon={<span />} label="delete" showInMenu />,
+          ]}
+        />,
+      );
+      const moreButton = screen.getByRole('menuitem', { name: 'more' });
+      fireEvent.mouseUp(moreButton);
+      fireEvent.click(moreButton);
 
+      const printButton = screen.queryByRole('menuitem', { name: 'print' });
+      expect(printButton).toHaveFocus();
+    });
+
+    it('should allow to navigate between actions using the arrow keys', () => {
+      render(
+        <TestCase
+          getActions={() => [
+            <GridActionsCellItem icon={<span />} label="print" />,
+            <GridActionsCellItem icon={<span />} label="delete" />,
+          ]}
+        />,
+      );
+      const firstCell = getCell(0, 0);
+      firstCell.focus();
       expect(getActiveCell()).to.equal('0-0');
+
+      fireEvent.keyDown(firstCell, { key: 'ArrowRight' });
+      const printButton = screen.getByRole('menuitem', { name: 'print' });
+      expect(printButton).toHaveFocus();
```

Comment:
Why do you have to press `arrowRight` to set the focus on the first button?

Reply:
The test component has 2 columns. First I focus on the first cell, then <kbd>ArrowRigth</kbd> to go to the next column.

Reply:
Ok, I missed that you added `id` column